### PR TITLE
fix(elasticsearch): verify whether mapping type is supported

### PR DESCRIPTION
**Pull request by darthf1**
| Q             | A
| ------------- | ---
| Branch?       | main
| Tickets       | https://github.com/api-platform/core/issues/4628
| License       | MIT
| Doc PR        | api-platform/docs#... 

supersedes https://github.com/api-platform/core/pull/4720

I propose to remove the mapping `type` from the search for certain ES versions, as is deprecated in ES 7 and removed in ES 8 (see [issue](https://github.com/api-platform/core/issues/4628)).

The `type` can still be passed in ES 7 but serves [no purpose](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/removal-of-types.html#_schedule_for_removal_of_mapping_types) anymore. Thats why the `type` will only be passed on ES 6 and lower so deprecations are not triggered.

### CODE REVIEW

**Comment by alanpoulain** [comment]
> ES 6 is [EOL](https://www.elastic.co/support/eol) since 2022-02-10, so I'm not sure you want to :) Testing on newer versions (like ES 8.) seems more beneficial (for other reasons than this MR).

I agree, could you add it to the CI?

File: src/Elasticsearch/Util/ElasticsearchVersion.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+declare(strict_types=1);
+
+namespace ApiPlatform\Elasticsearch\Util;
+
+use Elasticsearch\Client;
+
+class ElasticsearchVersion
+{
+    /**
+     * @see https://regex101.com/r/EWxpuO/2
```
[the remark this answers is not in the document]
Comment (darthf1) [reply src/Elasticsearch/Util/ElasticsearchVersion.php]: done

**Comment by darthf1** [comment]
> > ES 6 is [EOL](https://www.elastic.co/support/eol) since 2022-02-10, so I'm not sure you want to :) Testing on newer versions (like ES 8.) seems more beneficial (for other reasons than this MR).
> 
> I agree, could you add it to the CI?

Unfortunately I can't, I'm on gitlab myself so I don't know anything about github ci.

**Comment by soyuka** [comment]
This is a good fix thanks! I'll add the CI myself in another patch.

**Pull request merged**